mlx/coverity_services.py

```diff
@@ -1,253 +1,226 @@
-        if bool(idx):
-            num_snapshots = len(snapshots)
-            if idx < 0:
-                required = num_snapshots + idx + 1
-            else:
-                required = idx
-
-            if abs(required) > 0 and abs(required) <= num_snapshots:
-                # base zero
-                return snapshots[required - 1].id
-        return 0
-
-    def get_snapshot_detail(self, snapshot_id):
-        '''Get detailed information about a single snapshot'''
-        snapshot = self.client.factory.create('snapshotIdDataObj')
-        snapshot.id = snapshot_id
-        # return a snapshotInfoDataObj
-        return self.client.service.getSnapshotInformation(snapshot)
-
-    def get_checkers(self):
-        '''Get a list of checkers from the service'''
+    def retrieve_issues(self, filters):
+        """Retrieve issues from the server (Coverity Connect).
+
+        Args:
+            filters (json): The filters as json
+
+        Returns:
+            dict: The response
+        """
+        url = self.base_url.rstrip('/') + \
+            "/issues/search?includeColumnLabels=true&offset=0&queryType=bySnapshot&rowCount=-1&sortOrder=asc"
+        return self._post_request(url, filters)
+
+    def retrieve_column_keys(self):
+        """Retrieves the column keys and associated display names.
+
+        Returns:
+            list[dict]: A list of dictionaries where the keys of each dictionary are 'columnKey' and 'name'
+        """
+        if not self._columns:
+            url = f"{self.base_url.rstrip('/')}/issues/columns?queryType=bySnapshot&retrieveGroupByColumns=false"
+            self._columns = self._get_request(url)
+        return self.columns
+
+    def retrieve_checkers(self):
+        """Retrieve the list of checkers from the server.
+
+        Returns:
+            list[str]: The list of valid checkers
+        """
         if not self.checkers:
-            self.checkers = self.client.service.getCheckerNames()
+            url = f"{self.base_url.rstrip('/')}/checkerAttributes/checker"
+            checkers = self._get_request(url)
+            if checkers and "checkerAttributedata" in checkers:
+                self._checkers = [checker["key"] for checker in checkers["checkerAttributedata"]]
         return self.checkers
 
+    def _get_request(self, url):
+        """Make a GET request to the API.
+
+        Args:
+            url (str): The url to request data via GET request
+
+        Returns:
+            dict: the content of server's response
+
+        Raises:
+            requests.HTTPError
+        """
+        req = self.session.get(url)
+        if req.ok:
+            return json.loads(req.content)
+        else:
+            try:
+                message = json.loads(req.content)["message"]
+            except:
+                message = req.content.decode()
+            logger = getLogger("coverity_logging")
+            logger.warning(message)
+            return req.raise_for_status()
```

Comment:
```suggestion
        response= self.session.get(url)
        if response.ok:
            return response.json()
        try:
            err_msg= response.json()["message"]
        except (requests.exceptions.JSONDecodeError, KeyError):
            err_msg= response.content.decode()
        logger = getLogger("coverity_logging")
        logger.warning(err_msg)
        return response.raise_for_status()
```

`response` is a better name than `req`, which is short for request.

"There must be a better way!" See https://requests.readthedocs.io/en/latest/user/quickstart/#json-response-content

You can drop the indentation after `return`. You should always specify which exception types to catch. The variable name `message` is not descriptive enough imo. 

Do you know why `req.content` isn't always a JSON string? Is it a bug in the Coverity REST API?

Reply:
An example when you do not return a JSON string is when you use invalid credentials.
I don't think it's a bug.